File: PhotoAPI/Photo.Domain/DataTransferObjects/TextOnPhotoDTO.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace Photo.Domain.DataTransferObjects
{
    public class TextOnPhotoDTO
    {
        public string Language { get; set; }
        public double TextAngle { get; set; }
        public string Orientation { get; set; }
        public IList<Region> Regions { get; set; }
    }

    public class Region
    {
        public string BoundingBox { get; set; }
        public IList<Line> Lines { get; set; }
    }

    public class Line
    {
        public string BoundingBox { get; set; }
        public IList<Word> Words { get; set; }
    }

    public class Word
    {
        public string BoundingBox { get; set; }
        public string Text { get; set; }
    }
}

File: WebClient/lama/src/app/models/Photo/textOnPhotoDTO.ts
export interface TextOnPhotoDTO {
  language: string;
  textAngle: number;
  orientation: string;
  regions: Region[];
}

class Region {
  boundingBox: string;
  lines: Line[];
}

class Line {
  boundingBox: string;
  words: Word[];
}

class Word {
  boundingBox: string;
  text: string;
}
